src/network/networkd-link.c

```diff
@@ -218,6 +233,31 @@ static IPv6PrivacyExtensions link_ipv6_privacy_extensions(Link *link) {
         return link->network->ipv6_privacy_extensions;
 }
 
+static int link_enable_ipv6(Link *link) {
+        const char *p = NULL;
+        bool state = false;
+        int r;
+
+        if (link->flags & IFF_LOOPBACK)
+                return 0;
+
+        state = !link_ipv6_enabled(link);
+
+        p = strjoina("/proc/sys/net/ipv6/conf/", link->ifname, "/disable_ipv6");
+
+        r = write_string_file(p, one_zero(state), WRITE_STRING_FILE_VERIFY_ON_FAILURE);
+        if (r < 0)
+                log_link_warning_errno(link, r, "Cannot set IPv6 for interface: %m");
```

Comment:
This error will say "cannot set IPv6" even when *disabling* failed. Should be something like
```
"Cannot %s IPv6 for interface %s: %m", disabled ? "disable" : "enable", link->ifname
```